src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,86 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage {
+
+    WebDriver driver;
+
+    WebDriverWait wait;
+
+    Actions actions;
+
+    public BasePage(WebDriver givenDriver) {
+        driver = givenDriver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        actions = new Actions(driver);
+        PageFactory.initElements(driver, this);
+
+    }
+    // WEb Elements
+
+    // Using Page Factory;
+    @FindBy(css = "[type='email']")
+    WebElement emailField;
+    @FindBy(css = "[type='password']")
+    WebElement passwordField;
+    @FindBy(css = "[type='submit']")
+    WebElement submitBtn;
+    By soundBarVisualizer = By.cssSelector("[data-testid= 'sound-bar-play']");
+
+    public WebElement findElement(By locator) {
+        return wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
+
+    }
+   /* public void provideEmail(String email){
+        emailField.sendKeys(email);
+
+    }
+    public void providePassword(String password){
+        passwordField.sendKeys(password);
+    }
+    public void clickSubmit(){
+        submitBtn.click();
+    }
+    public void login(){
+        provideEmail("kaflimeerim@gmail.com");
+        providePassword("te$t$tudent");
+        clickSubmit();
+    }*/
```

Comment:
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.